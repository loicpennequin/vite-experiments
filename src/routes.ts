import { RouteRecordRaw } from 'vue-router';

export const routes: RouteRecordRaw[] = [
  {
    name: 'Home',
    path: '/',
    component: () => import('./pages/home/home.page.vue')
  },
  {
    name: 'Detail',
    path: '/:name',
    component: () => import('./pages/detail/detail.page.vue')
  },
  {
    name: 'Foo',
    path: '/foo',
    component: () => import('./pages/foo/foo.page.vue'),
    children: [
      {
        name: 'Bar',
        path: 'bar',
        component: () => import('./pages/foo/bar.page.vue')
      }
    ]
  }
];
