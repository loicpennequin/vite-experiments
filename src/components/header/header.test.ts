import { mount } from '@vue/test-utils';
import { createRouter, createWebHistory } from 'vue-router';
import { routes } from '../../routes';
import Header from './header.vue';

test('mount component', async () => {
  const router = createRouter({
    history: createWebHistory(),
    routes: routes
  });

  router.push('/');
  await router.isReady();
  expect(Header).toBeTruthy();

  const wrapper = mount(Header, { global: { plugins: [router] } });

  expect(wrapper.text()).toContain('Pok√©dex');
  expect(wrapper.html()).toMatchSnapshot();
});
