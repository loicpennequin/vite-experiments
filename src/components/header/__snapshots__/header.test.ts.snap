// Vitest Snapshot v1

exports[`Header Component > should mount component 1`] = `
<header flex="" items="center" p="3" bg="red-400" shadow="md">
  <h1 m="0" text="2xl"><a href="/" class="" no-underline="" color="inherit">Pok√©dex</a></h1>
</header>
`;
